
### Examining racial discrimination in the US job market

#### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

#### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes.

#### Exercise
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Discuss statistical significance.

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [2]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [3]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [4]:
# number of callbacks for white-sounding names
sum(data[data.race=='w'].call)

235.0

# Question 1 - What test is appropriate for this problem? Does CLT apply?

In [5]:
# let's examine the data set

data.describe()

education       ofjobs     yearsexp       honors    volunteer  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      3.618480     3.661396     7.842916     0.052772     0.411499   
std       0.714997     1.219126     5.044612     0.223601     0.492156   
min       0.000000     1.000000     1.000000     0.000000     0.000000   
25%       3.000000     3.000000     5.000000     0.000000     0.000000   
50%       4.000000     4.000000     6.000000     0.000000     0.000000   
75%       4.000000     4.000000     9.000000     0.000000     1.000000   
max       4.000000     7.000000    44.000000     1.000000     1.000000   

          military     empholes  occupspecific   occupbroad  workinschool  \
count  4870.000000  4870.000000    4870.000000  4870.000000   4870.000000   
mean      0.097125     0.448049     215.637782     3.481520      0.559548   
std       0.296159     0.497345     148.127551     2.038036      0.496492   
min       0.000000     0.000000       7.000000     1.000000      0.000000   
25%       0.000000     0.000000      27.000000     1.000000      0.000000   
50%       0.000000     0.000000     267.000000     4.000000      1.000000   
75%       0.000000     1.000000     313.000000     6.000000      1.000000   
max       1.000000     1.000000     903.000000     6.000000      1.000000   

          ...           educreq      compreq       orgreq        manuf  \
count     ...       4870.000000  4870.000000  4870.000000  4870.000000   
mean      ...          0.106776     0.437166     0.072690     0.082957   
std       ...          0.308866     0.496083     0.259649     0.275854   
min       ...          0.000000     0.000000     0.000000     0.000000   
25%       ...          0.000000     0.000000     0.000000     0.000000   
50%       ...          0.000000     0.000000     0.000000     0.000000   
75%       ...          0.000000     1.000000     0.000000     0.000000   
max       ...          1.000000     1.000000     1.000000     1.000000   

          transcom     bankreal        trade   busservice   othservice  \
count  4870.000000  4870.000000  4870.000000  4870.000000  4870.000000   
mean      0.030390     0.085010     0.213963     0.267762     0.154825   
std       0.171677     0.278932     0.410141     0.442847     0.361773   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     1.000000     0.000000   
max       1.000000     1.000000     1.000000     1.000000     1.000000   

           missind  
count  4870.000000  
mean      0.165092  
std       0.371308  
min       0.000000  
25%       0.000000  
50%       0.000000  
75%       0.000000  
max       1.000000  

[8 rows x 55 columns]

# Question 2: What are the null and alternate hypotheses?

Null Hypothesis: There is no difference in the call back proportion between the black sounding names and white sounding names.  P(black) = P(white)

Alternate Hypothesis: There is a difference in the call back proportion between black sounding names and white sounding names. P(black) <> P(white).

# Question 3: Compute margin of error, confidence interval, and p-value

In [52]:
# let's create data sets for the different races

black = data[data.race=='b']
white = data[data.race=='w']

# let's compute the sample proportions

black_p = black[black.call == 1].size / black.size
white_p = white[white.call == 1].size / white.size

difference = black_p - white_p

# let's compute the margin of error (aka standard error, aka sampling distribution)

black_moe = black_p * (1 - black_p) / black.size
white_moe = white_p * (1 - white_p) / white.size

moe = np.sqrt(black_moe + white_moe)

#p-value assuming 95% CI
pvalue = 1.96 * moe

print('The margin of error is', round(moe,4))
print('The pvalue is ', round(pvalue,4))
print('There is a 95% chance that the difference between the call back rates of black sounding and white sounding names,' \
      ,round(difference,4), 'is within', round(pvalue,4), 'of P(black) - P(white)')
print('The confidence interval is from ', round(moe - pvalue, 4), 'to', round(moe + pvalue,4))
print('Therefore we reject the null hypothesis because P(black) <> P(white), and conclude with 95% accuracy that ', \
     'there is a difference between the call back rates for black sounding and white sounding names, because the difference' \
      ' falls outside our confidence interval.')

The margin of error is 0.001
The pvalue is  0.0019
There is a 95% chance that the difference between the call back rates of black sounding and white sounding names, -0.032 is within 0.0019 of P(black) - P(white)
The confidence interval is from  -0.0009 to 0.0029
Therefore we reject the null hypothesis because P(black) <> P(white), and conclude with 95% accuracy that  there is a difference between the call back rates for black sounding and white sounding names, because the difference falls outside our confidence interval.


# Question 4: Discuss statistical significance

Statistical significance is a term that estimates the certainity of a statistical finding. In this example, it would describe how certain we are that there is a difference in the call back rates between black sounding and white sounding names.

In order to determine significance, a p-value is calculated. The p-value is the probability of observing an effect given that the null hypothesis is true. The null hypothesis is rejected when the p-value is less than our alpha level (in our case, 5% because that's what we picked - a 95% CI or a 5% alpha/significance level.

If the p-value is less than 5%, we can conclude that our result is statistically significant. The calculated p-value is 0.0019, which is < 0.05.